src/DependencyManagement/DependencyManager.cs

```diff
@@ -0,0 +1,223 @@
+            }
+
+            // Validate that version is in the correct format: 'MajorVersion.*'
+            if (!IsValidVersionFormat(version))
+            {
+                var errorMessage = String.Format(PowerShellWorkerStrings.InvalidVersionFormat, "MajorVersion.*");
+                throw new ArgumentException(errorMessage);
+            }
+
+            // Return the major version.
+            return version.Split(".")[0];
+        }
+
+        /// <summary>
+        /// Parses the given PowerShell (psd1) data file.
+        /// Returns a Hashtable representing the key value pairs.
+        /// </summary>
+        internal Hashtable ParsePowerShellDataFile(string functionAppRootPath, string fileName)
+        {
+            // Path to Requirements.psd1 file.
+            var requirementsFilePath = Path.Join(functionAppRootPath, fileName);
+
+            // Validate the file path.
+            if (!File.Exists(requirementsFilePath))
+            {
+                var errorMessage = String.Format(PowerShellWorkerStrings.FileNotFound, fileName, functionAppRootPath);
+                throw new ArgumentException(errorMessage);
+            }
+
+            // Try to parse the Requirements.psd1 file.
+            var ast = Parser.ParseFile(requirementsFilePath, out _, out ParseError[] errors);
+
+            if (errors?.Length > 0)
+            {
+                var stringBuilder = new StringBuilder();
+                foreach (var error in errors)
+                {
+                    stringBuilder.AppendLine(error.Message);
+                }
+
+                string errorMsg = stringBuilder.ToString();
+                throw new ArgumentException(string.Format(PowerShellWorkerStrings.FailToParseScript,
+                    RequirementsPsd1FileName, errorMsg));
+            }
+
+            var hashtableAst = ast.Find(x => x is HashtableAst, false);
+            var hashtable = default(Hashtable);
+
+            bool throwException = false;
+            try
+            {
+                hashtable = (Hashtable)hashtableAst?.SafeGetValue();
+            }
+            catch
+            {
+                throwException = true;
+            }
+
+            // Ensure that the hashtable is not null.
+            if (hashtable == null)
+            {
+                throwException = true;
+            }
+
+            if (throwException)
+            {
+                string errorMsg = string.Format(PowerShellWorkerStrings.InvalidPowerShellDataFile,
+                    RequirementsPsd1FileName);
+                throw new ArgumentException(errorMsg);
+            }
+
+            return hashtable;
+        }
+
+        /// <summary>
+        /// Validates the given version format. Currently, we only support 'Number.*'.
+        /// </summary>
+        internal bool IsValidVersionFormat(string version)
+        {
+            var pattern = @"^(\d){1,2}(\.)(\*)";
+            var regex = new Regex(pattern);

```

Comment:
Use `Regex.IsMatch` instead.

Reply:
Fixed. The code looks like this now:

var pattern = @"^(\d){1,2}(\.)(\*)";
return Regex.IsMatch(version, pattern);

Reply:
Fixed. The code looks like this now:
var pattern = @"^(\d){1,2}(.)(*)";
return Regex.IsMatch(version, pattern);